packages/main/src/RPA/Smartsheet.py

```diff
@@ -0,0 +1,1415 @@
+    specifying the ``native`` argument. The default will return a more
+    common Python object, such as a dictionary or list. The native object
+    is a class from the `smartsheet-python-sdk`_ library and will have
+    additional methods and attributes. The most important attributes
+    available for most native objects are (some may be unavailable
+    for some objects):
+
+    - ``id``: the unique identifier of the object
+    - ``name``: the name of the object
+    - ``title``: the title of a column
+    - ``permalink``: the URL to the object
+    """
+
+    # TODO: ADD LOGGING THROUGHOUT
+
+    # Only implemented includes are available, others exist for the API
+    SHEET_INCLUDES = {
+        "rowId": None,  # function must be set in __init__
+        "attachments": "attachments",
+        "attachmentFiles": None,  # function must be set in __init__
+        "discussions": "discussions",
+        "rowPermalink": "permalink",
+    }
+    """Dictionary of supported additional objects to ask for via ``include``
+    parameters for sheets. Key is the name of the option, value is the
+    API attribute or function to call.
+    """
+    ROW_INCLUDES = {
+        "rowId": None,  # function must be set in __init__
+        "attachments": "attachments",
+        "attachmentFiles": None,  # function must be set in __init__
+        "discussions": "discussions",
+        "rowPermalink": "permalink",
+        "filters": "filteredOut",
+    }
+    """Dictionary of supported additional objects to ask for via ``include``
+    parameters for rows. Key is the name of the option, value is the
+    API attribute or function to call.
+    """
+    ROW_EXCLUDES = [
+        "filteredOutRows",
+        "linkInFromCellDetails",
+        "linksOutToCellsDetails",
+        "nonexistentCells",
+    ]
+    """List of supported options to remove cells from rows."""
+    SEARCH_INCLUDES = {"favoriteFlag": None}
+    """Dictionary of supported additional objects to ask for via ``include``
+    parameters for searches. Key is the name of the option, value is the
+    API attribute or function to call.
+    """
+    SEARCH_SCOPES = [
+        "attachments",
+        "cellData",
+        "comments",
+        "folderNames",
+        "reportNames",
+        "sheetNames",
+        "sightNames",
+        "summaryFields",
+        "templateNames",
+        "workspaceNames",
+    ]
+
+    def __init__(self, access_token: str = None, max_retry_time: Union[str, int] = 30):
+        """If you do not initialize the library with an access token,
+        it will attempt to load the environment variable
+        ``SMARTSHEET_ACCESS_TOKEN``, otherwise, use the keyword
+        `Set access token`.
+
+        :param access_token: The access token created for your
+         Smartsheet user.
+        :param max_retry_time: Maximum time to allow retries of API
+         calls. Can be provided as a time string or int.
```

Comment:
What is our library convention though? I was told the single space was convention the last time I made a library. I beleive that comes from Mika, see the `JavaAccessBridge` library for his example.

Reply:
I'm working on it, indeed, we have mixed standards in the library, so feel free to go onwards with what you have already (as there isn't too much value on choosing one "standard" over another).

I personally used 4-spaces since the beginning since that's what Python usually uses in indenting everything.
- Stackoverflow [post](https://stackoverflow.com/a/50455731)
- [Example](https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/0.6.0/adal/authentication_context.py#L59-L85) from there
- Even Sphinx itself has [inconsistency](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) 😁 

So no strong opinion here, pick up what's faster to write and make the code/docstring readable.

Reply:
Although [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-py-function) are 4 spaces:

<img width="851" alt="Screenshot 2023-03-17 at 13 56 50" src="https://user-images.githubusercontent.com/709053/225824352-8ef024bd-9e0c-4b13-82e8-fec0252ba686.png">